forem/forem
# Load `post_sidebar` billboard asynchronously

## What type of PR is this? (check all applicable)


## Description

In [this PR](https://github.com/forem/forem/pull/19285) we setup the backend to be able to load the `post_comments`  display ad asynchronously and now in this PR, we do the same for `post_sidebar`. However, in this case we want to ensure that any ad on the article page can be loaded asynchronously. 

## Related Tickets & Documents


- Related Issue #https://github.com/forem/forem/issues/19227
- Closes #https://github.com/forem/forem/issues/19227

## QA Instructions, Screenshots, Recordings
- Create a Billboard on the admin to target `post_comments` and `post_sidebar`
- Load up an article and ensure that the ads are loading asynchronously.
- You can also open up the network panel to see the html response.
- Caching should occur when signed out.

https://www.loom.com/share/48f950c1fb694ff99a0200fa781478cb

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] No, and this is why: the tests were included in the previous PR and we currently can't really test the async ad display in js. Maybe there are potential test cases that I'm missing, open to feedback? 

## [optional] Are there any post deployment tasks we need to perform?

We need to monitor the caching, I'll be co-ordinating with Mac.

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/3o72EX5QZ9N9d51dqo/giphy.gif)
